pyuvdata/uvdata/initializers.py

```diff
@@ -0,0 +1,579 @@
+        the antenna_names, according to the antname_format. antenna_names need not be
+        provided if antenna_positions is a dict with string keys.
+    antenna_numbers : list of int, optional
+        List of antenna numbers. If not provided, antenna names will be used to form
+        the antenna_numbers, but in this case the antenna_names must be strings that
+        can be converted to integers. antenna_numbers need not be provided if
+        antenna_positions is a dict with integer keys.
+    blts_are_rectangular : bool, optional
+        Set to True if the time_array and antpair_array are rectangular, i.e. if
+        they are formed from the outer product of a unique set of times/antenna pairs.
+        If not provided, it will be inferred from the provided arrays.
+    data_array : ndarray of complex, optional
+        Array of data. If not provided, and ``empty=True`` it will be initialized to all
+        zeros. Otherwise a purely metadata-only object will be created. It must have
+        shape ``(Nblts, Nfreqs, Npols)``
+    flag_array : ndarray of bool, optional
+        Array of flags. If not provided, and ``empty=True`` it will be initialized to
+        all False. Otherwise a purely metadata-only object will be created. It must have
+        shape ``(Nblts, Nfreqs, Npols)``
+    nsample_array : ndarray of float, optional
+        Array of nsamples. If not provided, and ``empty=True`` it will be initialized to
+        all zeros. Otherwise a purely metadata-only object will be created. It must have
+        shape ``(Nblts, Nfreqs, Npols)``
+    flex_spw : bool, optional
+        Set to True if the frequency axis is not contiguous. Default is False.
+    history : str, optional
+        History string to be added to the object. Default is a simple string
+        containing the date and time and pyuvdata version.
+    instrument : str, optional
+        Instrument name. Default is the ``telescope_name``.
+    vis_units : str, optional
+        Visibility units. Default is 'uncalib'. Must be one of 'Jy', 'K str', or
+        'uncalib'.
+    antname_format : str, optional
+        Format string for antenna names. Default is '{0:03d}'.
+    empty : bool, optional
+        Set to True to create an empty (but not metadata-only) UVData object.
+        Default is False.
+    time_axis_faster_than_bls : bool, optional
+        Set to True if the time axis is faster than the baseline axis, under the
+        assumption that the blt axis is rectangular. This *sets* the blt order in the
+        case that unique times and unique baselines are provided, otherwise it is
+        inferred from the provided arrays.
+    phase_center_catalog : dict, optional
+        Dictionary of phase center information. Each key is a source id, and each
+        value is a dictionary. By default, a single phase center is assumed, at
+        zenith, assumed to be unprojected (i.e. drift scan).
```

Comment:
Ahh, I see. Yeah the phase center stuff is what I least understand out of all of this, and it seems the trickiest to get right. It also seems like choosing "unprojected" as the default could be highly contestible -- maybe choosing any default here is not useful. On the other hand, since it's not the easiest thing to set (from the user's point of view), I'd like to provide an easy default. This all might be easier if the `phase_center_catalog` was say a small class which had instantiation methods like `PhaseCenterCat.unprojected()` or `PhaseCenterCat.drift(az=..., za=...)`. The dictionary with all its required keys is a bit confusing.

Reply:
I could have a go at drafting something like this up, but obviously it would be API-breaking :/

Reply:
Moving to an object is the long term plan.